config/messaging.kafka.yml

```diff
@@ -1,9 +1,12 @@
 ---
 base: &base
-  hostname: localhost
+  host: localhost
   port: 9092
-  username: admin
-  password: smartvm
+  protocol: Kafka
+  encoding: json
+  sasl.mechanism: PLAIN
+  sasl.username: admin
+  sasl.password: smartvm
```

Comment:
Do we want to standardize this while we're at it?  and just make it `username` and `password` instead of the sasl. thing?

My concern is if we don't do it now, we probably never can because the migration will be a pain.

Reply:
Yeah we actually already pull username/password out of options if they're present so we could use just those https://github.com/ManageIQ/manageiq-messaging/blob/master/lib/manageiq/messaging/kafka/client.rb#L92-L93

Reply:
If we defaulted to `options[:'sasl.mechanism'] ||= 'PLAIN'` we could drop all `sasl.` options here.  We use `PLAIN` in all cases anyway.

If we added `:ssl => [true/false]` and `:ca_file => path` I think we could have all "generic" options without getting into librdkafka specifics, thoughts?

Reply:
E.g. this is what a production messaging.yml looks like now:
```yaml
  hostname: localhost
  port: 9093
  protocol: Kafka
  encoding: json
  sasl.mechanism: PLAIN
  sasl.username: admin
  sasl.password: v2:{ynR/dBrQ0iy/1fPSCyXWaA==}
  security.protocol: SASL_SSL
  ssl.ca.location: "/opt/kafka/config/keystore/ca-cert"
```

Suggesting:
```yaml
  hostname: localhost
  port: 9093
  protocol: Kafka
  encoding: json
  username: admin
  password: v2:{ynR/dBrQ0iy/1fPSCyXWaA==}
  ssl: true
  ca_file: "/opt/kafka/config/keystore/ca-cert"
```

Reply:
> If we added :ssl => [true/false] and :ca_file => path I think we could have all "generic" options without getting into librdkafka specifics, thoughts?

Love it.  Making it is great and then the protocol itself can translate (and then we don't need 2 different sample files either).

Reply:
One other very small thing, but why are the protocols uppercase?

Reply:
> Suggesting:

In your suggestion is hostname, but I think we want host.

Reply:
> One other very small thing, but why are the protocols uppercase?

Because manageiq-messaging passes that directly to `.const_get`, https://github.com/ManageIQ/manageiq-messaging/blob/master/lib/manageiq/messaging/client.rb#L42 we could probably `.classify` it there and be able to take `"kafka"` or `"Kafka"`

Reply:
> In your suggestion is hostname, but I think we want host.

Correct, I just copied the section and modified the ssl/sasl keys but I'll update the suggested to go back to `:host`

Reply:
https://github.com/ManageIQ/manageiq-messaging/pull/76